### ISSUE TITLE
Should be able to specify two naming styles for a given symbol

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#23707]
``` 
[*.{cs,vb}]
dotnet_naming_rule.constants_must_be_pascal_cased.symbols                                = constant_symbols
dotnet_naming_rule.constants_must_be_pascal_cased.style                                  = pascal_case_style,all_upper_style
dotnet_naming_rule.constants_must_be_pascal_cased.severity                               = warning

dotnet_naming_symbols.constant_symbols.applicable_kinds                                 = field
dotnet_naming_symbols.constant_symbols.required_modifiers                               = const
dotnet_naming_symbols.constant_symbols.applicable_accessibilities                       = *

# Styles
dotnet_naming_style.pascal_case_style.capitalization = pascal_case
dotnet_naming_style.all_upper_style.capitalization = all_upper
```

``` C#
    class Program
    {
        private const string fooBar = "fffo";
        private const string FooBar = "fffo";
        private const string FOO_BAR = "fffo";
    }

```

Expected: A single warning on "fooBar"
Actual: No warnings because the rule is invalid due to "pascal_case_style,all_upper_style"

**Comment by CyrusNajmabadi** [comment]
It seems strange to have a naming style that allows for two different styles.  What's the use case for this sort of thing?  Thanks! :)

**Comment by davkean** [comment]
I want constants to be allowed to be:

`private const string AppDesigner = nameof(AppDesigner)`
`private const int WM_PAINT = 0x000F`

But looks like I'm confused how this is enforced anyway: https://github.com/dotnet/roslyn/issues/23706

**Comment by sharwell** [comment]
@davkean I would expect naming rules to be either disabled within a `NativeMethods` class, similar to the heuristic in StyleCop. This would exclude `WM_PAINT` from consideration.

**Comment by davkean** [comment]
@sharwell In your projects perhaps.

**Comment by CyrusNajmabadi** [comment]
Closing out due to not hearing about this from ecosystem/community.
